common/lib/xmodule/xmodule/modulestore/tests/test_publish.py

```diff
@@ -139,3 +152,614 @@ def test_publish_draft_delete(self):
+            # Ensure that the vertical and its children are drafts in the exported OLX.
+            self.assertOLXIsDraft(block_list)
+            # Publish the vertical block.
+            self.publish('vertical', 'vertical00')
+            # Ensure that the vertical and its children are published in the exported OLX.
+            self.assertOLXIsPublished(block_list)
+
+    @ddt.data(*MODULESTORE_SETUPS)
+    def test_publish_multiple_verticals(self, modulestore_builder):
+        with self._setup_test(modulestore_builder, self.course_key) as (
+            self.root_export_dir, self.contentstore, self.store, self.course
+        ):
+            block_list_publish = (
+                ('vertical', 'vertical03'),
+                ('vertical', 'vertical04'),
+                ('html', 'unit06'),
+                ('html', 'unit07'),
+                ('html', 'unit08'),
+                ('html', 'unit09'),
+            )
+
+            block_list_untouched = (
+                ('vertical', 'vertical00'),
+                ('vertical', 'vertical01'),
+                ('vertical', 'vertical02'),
+                ('vertical', 'vertical05'),
+                ('vertical', 'vertical06'),
+                ('vertical', 'vertical07'),
+                ('html', 'unit00'),
+                ('html', 'unit01'),
+                ('html', 'unit02'),
+                ('html', 'unit03'),
+                ('html', 'unit04'),
+                ('html', 'unit05'),
+                ('html', 'unit10'),
+                ('html', 'unit11'),
+                ('html', 'unit12'),
+                ('html', 'unit13'),
+                ('html', 'unit14'),
+                ('html', 'unit15'),
+            )
+
+            # Ensure that both groups of verticals and children are drafts in the exported OLX.
+            self.assertOLXIsDraft(block_list_publish)
+            self.assertOLXIsDraft(block_list_untouched)
+
+            ##########################################
+            # Publish both vertical03 and vertical 04.
+            self.publish('vertical', 'vertical03')
+            self.publish('vertical', 'vertical04')
+            ##########################################
+
+            # Ensure that the published verticals and children are indeed published in the exported OLX.
+            self.assertOLXIsPublished(block_list_publish)
+            # Ensure that the untouched vertical and children are still untouched.
+            self.assertOLXIsDraft(block_list_untouched)
+
+    @ddt.data(*MODULESTORE_SETUPS)
+    def test_publish_single_sequential(self, modulestore_builder):
+        """
+        Sequentials are auto-published. But publishing them explictly publishes their children,
+        changing the OLX of each sequential - the vertical children are in the sequential post-publish.
+        """
+        with self._setup_test(modulestore_builder, self.course_key) as (
+            self.root_export_dir, self.contentstore, self.store, self.course
+        ):
```

Comment:
Given that this wraps every test, perhaps it would be better to call `__enter__` manually in `setUp`, and add `__exit__` as a cleanup method (again, in `setUp`). That way, the individual test cases don't need to each have the same boilerplate.